libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java

```diff
@@ -215,60 +234,12 @@ public RequestQueue getRequestQueue() {
 	 * @throws IOException 
 	 */
 	public RestResponse sendSync(RestRequest restRequest) throws IOException {
-		return sendSync(restRequest.getMethod(), restRequest.getPath(), restRequest.getRequestEntity(), restRequest.getAdditionalHttpHeaders());
+        Request request = buildRequest(restRequest);
+        Response response = okHttpClient.newCall(request).execute();
+        return new RestResponse(response);
 	}
 
 	/**
-	 * Send an arbitrary HTTP request synchronously, using the given method, path and httpEntity.
-	 * Note: Cannot be used by code on the UI thread (use sendAsync instead).
-	 * 
-	 * @param method				the HTTP method for the request (GET/POST/DELETE etc)
-	 * @param path					the URI path, this will automatically be resolved against the users current instance host.
-	 * @param httpEntity			the request body if there is one, can be null.
-	 * @return 						a RestResponse instance that has information about the HTTP response returned by the server.
-	 */
-	public RestResponse sendSync(RestMethod method, String path, HttpEntity httpEntity) throws IOException {
-		return sendSync(method, path, httpEntity, null);
-	}
-
-	/**
-	 * Send an arbitrary HTTP request synchronously, using the given method, path, httpEntity and additionalHttpHeaders.
-	 * Note: Cannot be used by code on the UI thread (use sendAsync instead).
-	 * 
-	 * @param method				the HTTP method for the request (GET/POST/DELETE etc)
-	 * @param path					the URI path, this will automatically be resolved against the users current instance host.
-	 * @param httpEntity			the request body if there is one, can be null.
-	 * @param additionalHttpHeaders additional HTTP headers to add the generated HTTP request, can be null.
-	 * @return 						a RestResponse instance that has information about the HTTP response returned by the server. 
-	 * 
-	 * @throws IOException
-	 */
-	public RestResponse sendSync(RestMethod method, String path, HttpEntity httpEntity, Map<String, String> additionalHttpHeaders) throws IOException {
-		return new RestResponse(httpStack.performRequest(method.asVolleyMethod(), path, httpEntity, additionalHttpHeaders, true));
-	}
-
-	/**
-	 * Uploads a new file to the server. This will create a new file at version 1.
-	 *
-	 * @param theFile The path of the local file to upload to the server.
-	 * @param name The name/title of this file.
-	 * @param title Title.
-	 * @param description Description.
-	 * @return The response received from the server.
-	 */
-	public RestResponse uploadFile(File theFile, String name, String title, String description) {
```

Comment:
Multipart post are properly supported by OkHttp, so uploadFile is back into FileRequests.java